Check for virt. env. initialization

tisnik opened the pull request.

tisnik committed:
Check for virt. env. initialization

tisnik committed:
Install Python virtual env

tisnik commented:
[test]

abs51295 commented on runtests.sh:
```diff
@@ -23,19 +32,27 @@ export PYTHONPATH=`pwd`/src
+        printf "${RED}Python virtual environment can't be initialized${NORMAL}"
+        exit 1
+    fi
     pip install -U pip
```
It's `pip` or `pip3` here?

tisnik replied:
it is correct, because the `pip` now surely runs in the virtual environment:

```
$ which pip
/usr/bin/pip

$ pip --version
pip 9.0.1 from /usr/lib/python2.7/site-packages (python 2.7)

$ virtualenv -p python3 venv && source venv/bin/activate
Running virtualenv with interpreter /usr/bin/python3
Using base prefix '/usr'
New python executable in /home/tester/venv/bin/python3
Also creating executable in /home/tester/venv/bin/python
Installing setuptools, pip, wheel...done.

$ which pip
~/venv/bin/pip

$ pip --version
pip 10.0.1 from /home/tester/venv/lib/python3.6/site-packages/pip (python 3.6)
```

tisnik replied:
let's check it :)

abs51295 reviewed (commented):
Please have a look at the comments.

abs51295 commented on runtests.sh:
```diff
@@ -23,19 +32,27 @@ export PYTHONPATH=`pwd`/src
 
+# now we are surely in the Python virtual environment
+
 `which pip3` install git+https://github.com/fabric8-analytics/fabric8-analytics-worker.git@561636c
```
If we are good with running `pip` here than let's change it to `pip` from `which pip3`. WDYT @tisnik ?

tisnik replied:
changed, and we are still green :)

tisnik committed:
Use
Usage:
  pip <command> [options]

Commands:
  install                     Install packages.
  download                    Download packages.
  uninstall                   Uninstall packages.
  freeze                      Output installed packages in requirements format.
  list                        List installed packages.
  show                        Show information about installed packages.
  check                       Verify installed packages have compatible dependencies.
  search                      Search PyPI for packages.
  wheel                       Build wheels from your requirements.
  hash                        Compute hashes of package archives.
  completion                  A helper command used for command completion.
  help                        Show help for commands.

General Options:
  -h, --help                  Show help.
  --isolated                  Run pip in an isolated mode, ignoring environment variables and user configuration.
  -v, --verbose               Give more output. Option is additive, and can be used up to 3 times.
  -V, --version               Show version and exit.
  -q, --quiet                 Give less output. Option is additive, and can be used up to 3 times (corresponding to WARNING, ERROR, and
                              CRITICAL logging levels).
  --log <path>                Path to a verbose appending log.
  --proxy <proxy>             Specify a proxy in the form [user:passwd@]proxy.server:port.
  --retries <retries>         Maximum number of retries each connection should attempt (default 5 times).
  --timeout <sec>             Set the socket timeout (default 15 seconds).
  --exists-action <action>    Default action when a path already exists: (s)witch, (i)gnore, (w)ipe, (b)ackup, (a)bort.
  --trusted-host <hostname>   Mark this host as trusted, even though it does not have valid or any HTTPS.
  --cert <path>               Path to alternate CA bundle.
  --client-cert <path>        Path to SSL client certificate, a single file containing the private key and the certificate in PEM format.
  --cache-dir <dir>           Store the cache data in <dir>.
  --no-cache-dir              Disable the cache.
  --disable-pip-version-check
                              Don't periodically check PyPI to determine whether a new version of pip is available for download. Implied with
                              --no-index. from venv

abs51295 reviewed (approved):
LGTM.
